### ISSUE TITLE
some questions about training

### ISSUE THREAD

**Issue by ginvermouth** [jackyko1991/vnet-tensorflow#1]
hi ，thank you for sharing your repo.

1. can i use the same data for testing and evaluation？

2. how to train for multi class?   For 2 classes , we assign 0 for background and 1 for foreground. but  for 3 classes, i.e. background, organ, tumor ,what change should i take?

**Comment by jackyko1991** [comment]
1. Yes you can do that. But you may suffering overfitting problem and cannot properly evalutate your trained model.
Scikit Learn do have some good tools to to the training data separation. You can check it out 
[here](http://scikit-learn.org/stable/modules/generated/sklearn.model_selection.train_test_split.html)

2. Theoractically it can be arbitary labels. But I recommends consecutive unsigned integeres like {0,1,2} for 3 classes segmentations. You need to change 
`logits = VNet.v_net(images_placeholder,input_channels = input_batch_shape[4], output_channels =2)`
with `output_channels = 3` in `train.py`

I haven't really tested on this because I am lacking data. Current training uses cross entropy and faciliates multiple label classification, which is a bit different from DICE loss as described in the original papaer.

**Comment by ginvermouth** [comment]
@jackyko1991 Thank you very much for your reply. Sorry to border you again. I get some results but it seems a little weird. the result should be like this:
![image](https://github.com/ginvermouth/png/blob/master/IMG11.jpg)
but it shows like this:
![image](https://github.com/ginvermouth/png/blob/master/res1.png)
could you give me some advice?

**Comment by jackyko1991** [comment]
@ginvermouth how many epochs have you trained?

It takes me quite a long time to train a satisfactory result. Also Some preprocessing may help to improve the training result

**Comment by jackyko1991** [comment]
The two images are in different sizes (I am not sure if they are the same in .nii format). Pixel size do play a key role in training the model.

**Comment by ginvermouth** [comment]
@jackyko1991  really thanks.  sorry the results were captured by screenshot.  Only 15 cases were used as train data.   

I used the command like below: 
print(ct_array.shape,np.max(ct_array),np.min(ct_array))
print(seg_array.shape,np.max(seg_array),np.min(seg_array))

and  get 
(107, 256, 256) 200.0 -120.0  
(107, 256, 256) 1.0 0.0

because the CT volumes were used in my experiment,  I set the ct_array to HU value.  Is that ok?
and my jaccard loss:
![img](https://github.com/ginvermouth/png/blob/master/jac_los.png)
accuracy: 
![img](https://github.com/ginvermouth/png/blob/master/accuracy.png)

**Comment by jackyko1991** [comment]
15 cases is too few for good result. In my experiences, 50 cases would give a reasonable result, while > 100 could have DICE > 90%. Accuracy is largely affected by 0 label (background) so it is not a very good benchmarking index, use DICE or JACCARD instead.

[LiTS](https://competitions.codalab.org/competitions/17094#learn_the_details) do provide a big liver dataset with good labels. I am not sure if the data is still available.

For liver segmentations, I suggest you to perform volume cropping and let liver tissues occupies as much space as possible. Some previously know knowledge, e.g. extracting body part and excluding air voxels, could largely remove irrelevant and benefiting the training.
